### DISCUSSION TITLE
How to use WinMerge in P4V as Merge tool (not Diff)?

### DISCUSSION THREAD

**Discussion by thomas-seiler-bl** [WinMerge/winmerge#1762] [Q&A]
Hi, 
how is the syntax when using WinMerge as 3-way merge tool in P4V? I have the arguments %b %1 %2 and %r but not sure in which order I should use them and if additional ones are recommended.

**Comment by sdottaka** [comment]
WinMerge is not better than p4merge for 3-way merge, so I would recommend using p4merge, but if you do use WinMerge, the command line would look like this (I have never used Perforce, so I have not tried the following command line to see if it works correctly)

- When merging on the middle pane

`"C:\Program Files\WinMerge\WinMergeU.exe" /e /am /ub /fm /wl /wr %1 %b %2 /o %r`

- When merging on the right pane

`"C:\Program Files\WinMerge\WinMergeU.exe" /e /ar /ub /fr /wl /wm %b %1 %2 /o %r`

See below for command line options:
https://manual.winmerge.org/en/Command_line.html
